docs/application/dotnet/guides/opentk/work-with-skiasharp.md

```diff
@@ -0,0 +1,183 @@
+
+In OpenTK application, you can use SkiaSharp to draw geometries, texts or images. For more information about the SkiaSharp APIs, please reference to [SkiaSharp API document](https://docs.microsoft.com/en-us/dotnet/api/skiasharp?view=skiasharp-1.60.3).
+
+To draw something with SkiaSharp in OpenTK application, it following some general guidelines:
+
+-   Allocate a memory block as a canvas for SkiaSharp.
+-   Create SKCanvas on this memory block and render text/image you want on this SKCanvas with SkiaSharp.
+-   Create 2D texture from this memory block.
+-   Draw 2D texture with graphics APIs of OpenTK.
+
+## Allocate memory block
+
+Allocate a specific size memory block. The size of memory block must be same with the size of the canvas you want.
+```C#
+private IntPtr pBitMap;                   // memory handle
+private int rowByte;                      // bytes of one row 
+private int bitmapHeight, bitmapWidth;    // height and width of the canvas
+
+private void CreateBitmap()
+{
+    bitmapHeight = (int)(skiaPercent * mainWindow.Height);
+    bitmapWidth = (int)(skiaPercent * mainWindow.Width);
+    pBitMap = Marshal.AllocHGlobal(bitmapWidth * bitmapHeight * 4);     // allocate a memory block
+
+    rowByte = bitmapWidth * 4;
+}
+
+private void FreeBitmap()
+{
+    if (pBitMap != IntPtr.Zero)
+    {
+        Marshal.FreeHGlobal(pBitMap);   // free the allocated memory block
+        pBitMap = IntPtr.Zero;
+    }
+}
+```
+
+## Create SKCanvas on memory block
+
+Create SKSurface on the memory block, and then you can get the SKCanvas from the SKSurface.
+```C#
+private SKSurface surface;
+private SKCanvas canvas;
+
+private void CreateSKCanvas()
+{
+    GL.PixelStore(All.UnpackAlignment, 4);
+    
+    // create the surface
+    var info = new SKImageInfo(bitmapWidth, bitmapHeight, SKImageInfo.PlatformColorType, SKAlphaType.Premul);
+    surface = SKSurface.Create(info, pBitMap, rowByte);
+    if (surface != null)
+    {
+        canvas = surface.Canvas;
+    }
+}
+```
+
+## Draw text on the SKCanvas
+
+```C#
+private void DrawTextBySkiaSharp(SKCanvas canvas, int canvasWidth, int canvasHeight)
+{
+    canvas.DrawColor(SKColors.White);                                   // clear the canvas as white
+
+    using (var paint = new SKPaint())
+    {
+        paint.TextSize = 64.0f;
+        paint.IsAntialias = true;
+        paint.Color = SKColors.Red;
+        paint.IsStroke = false;
+        paint.TextAlign = SKTextAlign.Center;
+
+        canvas.DrawText("SkiaSharp", width / 2f, 144.0f, paint);        // Draw text 'SkiaSharp' on the canvas
+    }
+}
+
+private void DrawTextOnSkCanvas()
+{
+    if (canvas != null)
+    {
+        DrawTextBySkiaSharp(canvas, bitmapWidth, bitmapHeight);
+        canvas.Flush();
+    }
+}
+```
+
+## Create 2D texture from memory block.
+
+Call `GL.TexImage2D` of `OpenTK.Graphics.ES20` to generate 2D texture from the memory block.
```

Comment:
Please change to:
Call `GL.TexImage2D` of `OpenTK.Graphics.ES20` to generate 2D texture from the memory block: